libfwupdplugin/fu-dpaux-device.c

```diff
@@ -75,6 +75,10 @@ fu_dpaux_device_setup(FuDevice *device, GError **error)
 	guint8 buf[FU_STRUCT_DPAUX_DPCD_SIZE] = {0x0};
 	g_autoptr(GByteArray) st = NULL;
 
+	/* already done */
+	if (priv->dpcd_dev_id != NULL)
```

Comment:
Is this right? I would think for rescan you need to clear it and do it again.

Reply:
You're completely right -- What about adding a new vfunc that invalidate can call so that we can clear things like this?

Reply:
I did that, which looks much nicer.